src/components/Contributors.tsx

```diff
@@ -0,0 +1,167 @@
+import { Component } from 'react';
+import * as React from 'react';
+
+interface IContributorsState {
+    key: number;
+    photo: string;
+    githubPage: string;
+    githubName: string;
+    commits: number;
+}
+
+interface IReposState {
+    key: number;
+    name: string;
+    description: string;
+    link: string;
+}
+
+interface IContributorsArrayState {
+    ignoreRepos: string[];
+    ignoreContributors: string[];
+    repos: IReposState[];
+    contributors: IContributorsState[];
+}
+
+class Contributors extends Component<{}, IContributorsArrayState> {
+    
```

Comment:
States suggest the data is mutable during the user's interaction with the page. Can we classify the above structs as `type` instead, for example 
```typescript
type Contributor = {
    ignoreRepos: string[],
    ignoreContributors: string[],
    repos: ReposType[],
    contributors: ContributorsType[]
};

type Repo {
    key: number,
    name: string,
    description: string,
    link: string
};
```

and add the following as constants
```suggestion
private ignoreRepos = ["sicp", "sharedb-ace-backend", "assessments", "tools", "source-academy2"];
private ignoreContributors: ["dependabot[bot]", "dependabot-preview[bot]"],
private repos: Repo[],
private contributors: Contributor[]
```

Reply:
I have changed the structs from interface to type as suggested in my latest commit. Yet to figure out how to change parts of the code if I add ignoreRepos, ignoreContributors, repos, and contributors as constants instead of managing them within the state.

Reply:
You can directly refer to the constants like `this.ignoreRepos` and `this.repos = ...`.

We shouldn't store them as states if they need not be.

Reply:
Yes agreed, I have made the necessary changes in the latest commit.